Finish refactor src

MrWittman612 opened the pull request:
## Related Issue

Finished up the refactor

## Description
made folders and exported component form index.js
1. MTableBody 
2. MTableBodyRow
3. MTableEditField


## Related PRs

List related PRs against other branches:

| branch              | PR       |
| ------------------- | -------- |
| other_pr_production | [link]() |
| other_pr_master     | [link]() |

## Impacted Areas in Application

All

## Additional Notes

This is bad but you should go through it with a fine-tooth comb. Let me when you have questions

FalkF committed:
fix(DataManager): set new groupsIndex in sortData()

MrWittman612 committed:
refactored MTableBodyRow class component to functional component create MTableBodyRow folder export MTableBodyRow from index.js

MrWittman612 committed:
refactored MTableBodyRow class component to functional component create MTableBodyRow folder export MTableBodyRow from index.js

MrWittman612 committed:
uncommit console log that was braking test el.find not a function I'm on windows

MrWittman612 committed:
uncommit console log that was braking test el.find not a function I'm on windows

MrWittman612 committed:
deleted uncommited class component

MrWittman612 committed:
deleted uncommited class component

MrWittman612 committed:
uncommitted import

MrWittman612 committed:
uncommitted import

MrWittman612 committed:
got ride of the render function returned just the return lifted logic and props destructor to the top of the component

MrWittman612 committed:
got ride of the render function returned just the return lifted logic and props destructor to the top of the component

MrWittman612 committed:
passed options to CommonValues elementSize function not props

MrWittman612 committed:
passed options to CommonValues elementSize function not props

MrWittman612 committed:
MTableBody fully refactored to functional component

MrWittman612 committed:
MTableBody fully refactored to functional component

MrWittman612 committed:
fully refactored MTableBody to functional component

MrWittman612 committed:
fully refactored MTableBody to functional component

MrWittman612 committed:
converted MTableEditCell class component to function component

MrWittman612 committed:
converted MTableEditCell class component to function component

MrWittman612 committed:
messed thing up refactored wrong component in wrong folder fixed now

MrWittman612 committed:
messed thing up refactored wrong component in wrong folder fixed now

MrWittman612 committed:
converted MTableEditField class component to functional component

MrWittman612 committed:
got ride of getProps fnc destructered props

matthewoestreich committed:
update imports

matthewoestreich committed:
Update data-manager.js

matthewoestreich committed:
Merge pull request #74 from FalkF/2258-fix-groupsIndex-core-rebase

Fix expanded multi-level grouping collapsing on row selection

matthewoestreich committed:
Release 2.3.8

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
fix export issue

matthewoestreich committed:
Release 2.3.9

matthewoestreich committed:
update

trying to make things more legible

matthewoestreich committed:
use dynamic imports for exporters to limit bloat; fix formatting; add demo

matthewoestreich committed:
Release 2.3.10

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Release v2.3.11

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Release v2.3.12

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Release v2.3.13

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Release v2.3.14

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Release v2.3.15

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Release v2.3.16

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Release v2.3.17

matthewoestreich committed:
Update publish.yml

matthewoestreich committed:
Release v2.3.18

matthewoestreich committed:
update exporters

matthewoestreich committed:
Release v2.3.19

Domino987 committed:
Fix for #80

Due to the useEffect, onBulkEditRowChanged was called on render, since the previous comparison of the dependency array was not there (before the render).
So useEffects also run on render, which populates the bulkEditData object for all rows, instead of just the changed rows.

It should only be called onChange.

Domino987 committed:
Forward ref MTableEditRow

Domino987 committed:
Add ref to container

matthewoestreich committed:
Merge pull request #81 from material-table-core/Domino987-bulk-edit-fix

Fix for #80

Domino987 committed:
Add forwardRef to action

Domino987 committed:
Add forwardRef to actions

Domino987 committed:
Add forwardRef to TableCell

Domino987 committed:
Add forwardRef to EditTableCell

Domino987 committed:
Add forwardRef to booleanfield

Domino987 committed:
Add forwardRef to CurrencyField

Domino987 committed:
Add forwardRef to Datefield

Domino987 committed:
Merge pull request #83 from material-table-core/master

Domino987 committed:
Add forwardRef to DateTimefield

Domino987 committed:
Add forwardRef to EditField

Domino987 committed:
Add forwardRef to Lookupfield

Domino987 committed:
Add forwardRef to Textfield

Domino987 committed:
Add forwardRef to timefield

Domino987 committed:
Add forwardRef to boolean

Domino987 committed:
add forwardRef to datefilter

Domino987 committed:
Add forwardRef to defaultfilter

Domino987 committed:
Add forwardRef to Filter

Domino987 committed:
Add forwardRef to FilterRow

Domino987 committed:
Add forwardRef to filterrow

Domino987 committed:
Add forwardRef to Groupbar

Domino987 committed:
Add forwardRef to header

Domino987 committed:
Add forwardRef to groupRow

Domino987 committed:
Add forwardRef to pagination

Domino987 committed:
Add forwardRef to stepped pagination

Domino987 committed:
Add forwardRef to toolbar

Domino987 committed:
Add forwardRef to overlayerror

Domino987 committed:
Add forwardRef to overlaylaoding

Domino987 committed:
Fix for textfield name

matthewoestreich committed:
Update index.js

FIxes error "MTablePaginationInner does not exist"

DominikEngel committed:
Fix small ui error

matthewoestreich committed:
update exporters

matthewoestreich committed:
Release 2.3.20

Domino987 committed:
Fix typo

peacechen commented:
@MrWittman612 
Thanks for contributing to this project. I like the functional refactor. My only concern is what bugs may be introduced from this. How much testing has this had?

matthewoestreich committed:
Update package.json

matthewoestreich committed:
Update package.json

matthewoestreich committed:
Release 2.3.21

matthewoestreich committed:
improve exporters

matthewoestreich committed:
Release 2.3.22

MrWittman612 commented:
> @MrWittman612
> Thanks for contributing to this project. I like the functional refactor. My only concern is what bugs may be introduced from this. How much testing has this had?

As far as testing. I have run it on my box. It ran with no console errors. What bugs do You Think it will cause? Do you have any suggestions on how to further refactor?

peacechen commented:
Since there are no automated tests, we have to rely on manual testing for now. That tends to leave a lot of holes that don't get discovered until users deploy it in their apps.

@oze4 Given the above, I recommend placing this refactor in a branch and publish on npm under an alpha tag. That will allow people to start using it while not breaking production apps.

matthewoestreich commented:
We've already refactored nearly all components into functional components so, while a good suggestion, it's a little too late and would be kind of pointless. 

This is on hold for now anyhow...we would like to get the code base stable before introducing any more changes (we really need contributors, there's a lot that needs to be done and not very many people).

matthewoestreich commented:
Also worth noting an alpha branch will be created once things are stable and we have introduced the latest version of material-ui.

hbermani committed:
Added CustomExport funcationality and updated Demo

matthewoestreich committed:
Update README.md

matthewoestreich committed:
update MTableBodyRow

See #87

matthewoestreich committed:
Release 2.3.23

hbermani committed:
Updated customExport prop to return unknown

hbermani committed:
Added comments to explain code

hbermani committed:
Added comments for further explanation

MrWittman612 commented:
Has anyone looked at this should I just remove it since it doesn't appear that it will be used?

matthewoestreich commented:
It will be used. But again, we have had other priorities and don't plan on merging this yet. I personally not had time to look at this yet. Apologies. We will get to it ASAP.

matthewoestreich committed:
add types for exporters/clean up build process for exporters

matthewoestreich committed:
Release v2.3.24

Domino987 committed:
Merge branch 'master' into master

Domino987 committed:
Merge pull request #86 from hbermani/master

Added CustomExport functionality and updated Demo

Domino987 committed:
Merge pull request #82 from material-table-core/Domino987-forwardRef

matthewoestreich committed:
Update utils.js

see #90

matthewoestreich committed:
Release v2.3.25

matthewoestreich committed:
remove exporters since we created a new package for that

matthewoestreich committed:
See #90

matthewoestreich committed:
Release v2.3.26

Domino987 committed:
Update CellStyle Types

Domino987 committed:
Merge pull request #93 from material-table-core/Domino987-cellstyle

Update CellStyle Types

Domino987 committed:
Set CellStyle data type to any

mdpx committed:
fix #94

mdpx committed:
small code optimization in m-table-body

vmihalachi committed:
Removed ID - prevented multiple tables

Having the same ID prevented was shows as an error by https://accessibilityinsights.io/ when there were multiple tables on same page. 

I checked and didn't find any other reference to this id in code.

Commit:
closes #96: missing colSpan prop for MTableCell

matthewoestreich committed:
Update README.md

matthewoestreich committed:
Merge pull request #99 from vmihalachi/patch-1

Removed ID - prevented having multiple tables

matthewoestreich committed:
Merge pull request #100 from ivoberger/bugfix/96-missing-colspan-prop

closes #96: missing colSpan prop for MTableCell

matthewoestreich committed:
Merge pull request #98 from mdpx/fix/actions-row-break

Fix/actions row break

matthewoestreich committed:
Release 2.3.27

peacechen committed:
Fix row render of built-in action buttons and user-defined free actions

matthewoestreich committed:
Merge pull request #112 from peacechen/master

Fix row render of built-in action buttons and user-defined free actions

matthewoestreich committed:
Release v2.3.27

Domino987 committed:
Make params of onQueryChange optional

This fixes #110 to make the onQueryChange parameters optional.

Domino987 committed:
Access bulk edit on render

Domino987 committed:
Pass bulk edit rows to edit row

Domino987 committed:
Pass bulk changes to body

Domino987 committed:
Merge pull request #114 from material-table-core/RestoreEditValuesOnPage

Access bulk edit on render

Commit:
Use the current sort order when switching sorted column if defined

matthewoestreich committed:
Merge pull request #113 from material-table-core/Domino987-onQueryChange

Make params of onQueryChange optional

peacechen committed:
Add prop options.showDetailPanelIcon

Domino987 committed:
Merge pull request #136 from peacechen/master

Add prop options.showDetailPanelIcon

peacechen committed:
Add prop renderSummaryRow

peacechen committed:
Add summary cell styling

Domino987 committed:
Merge pull request #137 from peacechen/master

Add prop renderSummaryRow

matthewoestreich committed:
Merge pull request #126 from MagicAardvark/master

Use the current sort order when switching sorted column if defined

vdh committed:
Fixed bad icon types

The typings on the icons are incorrect and should match what `forwardRef` returns

This is based on details from my issue on the original repo: https://github.com/mbrn/material-table/issues/1553

matthewoestreich committed:
Release v2.3.29

peacechen committed:
Fix TS definition for renderSummaryRow (#140)

peacechen committed:
Fix bulkEditChangedRows prop type (#142)

Domino987 committed:
Merge pull request #141 from peacechen/master

Fix TS definition for renderSummaryRow (#140), bulkEditChangedRows propType (#142)

peacechen committed:
Update MTableHeader to support showDetailPanelIcon

peacechen committed:
Account for action icons in summary row

peacechen committed:
Add prop `onBulkEditOpen`.

Remove `bulkEditChangedRows` passed to TableRow in MTableEditRow (caused warning)

peacechen committed:
Edit row should render empty action icons instead of a TableCell.

timkrase committed:
Add action type definitions

matthewoestreich commented:
This is still on our radar. Apologies - I've gotten a little busy, so things have been moving slow. (I know @Domino987 has also been busy).

Thank you for your time/help!

Domino987 committed:
Merge pull request #154 from timkrase/fix/actionTypes

Add action type definitions & prop-types for function actions with position

Domino987 commented:
This looks great. Now we only need to remove the merge conflicts.

Domino987 committed:
Remove deep level access of styles

Domino987 committed:
Hide export menu on click

Domino987 committed:
Pass selectionProps optional to changeAllSelected

Domino987 committed:
Pass selectionProps to changeAllSelected

Domino987 committed:
Merge pull request #160 from material-table-core/SelectAllNotDisabled

Select all not disabled

Domino987 committed:
Merge pull request #159 from material-table-core/HideExportMenuOnClick

Hide export menu on click

Domino987 committed:
Merge pull request #157 from material-table-core/TopLevelAccess

Remove deep level access of styles

matthewoestreich committed:
Update package.json

Domino987 committed:
Merge pull request #144 from peacechen/master

Update MTableHeader to support showDetailPanelIcon

peacechen committed:
Fix hiddenByColumnsButton (#117)

Domino987 committed:
Merge pull request #139 from vdh/patch-1

Fixed bad icon types

matthewoestreich committed:
update demo

Domino987 committed:
Merge pull request #164 from peacechen/master

Fix hiddenByColumnsButton (#117)

matthewoestreich committed:
fix issue with hiding cols in columns button

matthewoestreich committed:
Release v2.3.31

matthewoestreich committed:
Update CONTRIBUTING.md

matthewoestreich committed:
Update CONTRIBUTING.md

matthewoestreich committed:
Create PUBLISHING.md

matthewoestreich committed:
Update CONTRIBUTING.md

matthewoestreich committed:
Update PUBLISHING.md

matthewoestreich committed:
update demo

matthewoestreich committed:
change import fast-deep-equal to fast-deep-equal/react

matthewoestreich committed:
Update package.json

matthewoestreich committed:
Update package.json

Domino987 committed:
Merge pull request #166 from material-table-core/fix-peer-deps-react

Fix peer deps react

peacechen committed:
Right-align numeric inputs (#149)

matthewoestreich committed:
Update index.js

matthewoestreich committed:
Release v2.3.32

matthewoestreich committed:
Update package.json

matthewoestreich committed:
Merge pull request #172 from material-table-core/fix_date-io/core_version

Fix @date-io/core Version

matthewoestreich committed:
Release v2.3.33

pouryapb committed:
Column Resizable fixed

Domino987 committed:
Merge pull request #165 from peacechen/master

Right-align numeric inputs (#149)

pouryapb committed:
better solution for resizable column fix

Domino987 committed:
Merge pull request #175 from pouryapb/master

Column Resizable fixed

matthewoestreich committed:
see #176

fixes it

matthewoestreich committed:
Release v2.3.34

matthewoestreich committed:
fix issue with showDetailPanelIcon and bulk edit

matthewoestreich committed:
Merge pull request #178 from material-table-core/fix-detail-panel-showDetailPanelIcon-with-bulk-edit

fix issue with showDetailPanelIcon and bulk edit

matthewoestreich committed:
Release v2.3.35

mdpx committed:
fix #122

pouryapb committed:
added mouseEnter and mouseLeave listeners to action

pouryapb committed:
typo!

matthewoestreich committed:
spread all handlers

add handlers prop, so consumer can supply any event they wish, not just onMouseEnter or onMouseLeave. Keeping the legacy 'onClick' prop (the one outside of 'handlers') for legacy reasons

matthewoestreich committed:
Merge branch 'master' into master

matthewoestreich committed:
Merge pull request #181 from pouryapb/master

Added onMouseEnter and onMouseLeave listeners to action

matthewoestreich committed:
Release v2.3.36

mdpx committed:
Merge remote-tracking branch 'upstream/master'

Domino987 committed:
Create helper.d.ts

Domino987 committed:
Update index.d.ts

Domino987 committed:
Merge pull request #184 from material-table-core/HandlerProps

Add types of the new  action handlers

Domino987 committed:
docs: add changelog

Domino987 committed:
updated CHANGELOG.md

Domino987 committed:
Release v2.3.37

Domino987 committed:
updated CHANGELOG.md

Domino987 committed:
Release v2.3.38

matthewoestreich committed:
Update PUBLISHING.md

matthewoestreich committed:
Merge pull request #186 from material-table-core/Add-Changelog

Add changelog

matthewoestreich committed:
defaultGroupOrder possible fix

matthewoestreich committed:
Update index.js

matthewoestreich committed:
modify fix for defaultGroupOrder warning

matthewoestreich committed:
Merge pull request #188 from material-table-core/defaultGroupOrder-issues

Resolve defaultGroupOrder PropTypes Warning

matthewoestreich committed:
Fix wrong version

matthewoestreich committed:
updated CHANGELOG.md

matthewoestreich committed:
Update package-lock.json

matthewoestreich committed:
updated CHANGELOG.md

matthewoestreich committed:
Release v2.3.37

matthewoestreich committed:
Update package.json

matthewoestreich committed:
Update package.json

Commit:
added option to choose sort order on column switch between keep same order or reset to asc

Domino987 committed:
Removal of fade since its deprecated in v5

Domino987 committed:
Remove fade of loading

matthewoestreich committed:
Merge pull request #193 from material-table-core/Remove-fade

Remove fade

Commit:
reverting changes not concerned by the PR

Commit:
making sort order computation function for more maintainability and renaming prop keepSortOrderOnColumnSwitch to keepSortDirectionOnColumnSwitch

MrWittman612 commented:
sorry for the slow response on this if you need me to take any action on this on my part just let me know.

Domino987 committed:
Merge pull request #192 from LeComptoirDesPharmacies/enable-choosing-sort-order-when-switching-sorted-column

added option to choose sort order on column switch between keep same order or reset to asc…

Domino987 committed:
Merge pull request #180 from mdpx/master

Domino987 commented on src/components/MTableBody/index.js:
```diff
@@ -0,0 +1,353 @@
+  renderData,
+  initialFormData
+}) {
+  const renderDataC = renderData;
```
What is the purpose of this copy here?

MrWittman612 replied:
there was a naming conflict so I had to rename the function? But I do agree that there is more work that could be done

Domino987 replied:
You can simply rename it in the destructure which makes it clearer {oldName: newName} to avoid name conflicts.

Domino987 commented on src/index.js:
```diff
@@ -29,7 +29,7 @@ export default withStyles(styles, { withTheme: true })((props) => (
 
 export {
-  MaterialTable as MTable,
+  // MaterialTable as MTable,
```
Why is this export removed?

matthewoestreich replied:
I can't remember if I did this or not, but from what I can tell MaterialTable isn't exported from './components' - even outside of that, wouldn't it be redundant since MaterialTable is the default export? Why have it as default and included in named exports?

It is not currently being exported as of v2.3.39 so it doesn't look like it's causing issues.

Domino987 replied:
Ok sounds good.

Domino987 commented on src/components/MTableEditCell/index.js:
```diff
@@ -1,31 +1,15 @@
 
 function MTableEditCell(props) {
-  const [state, setState] = useState(() => ({
+  const [state, setState] = React.useState(() => ({
```
Why is here a function generation instead of just the object itself?

Domino987 commented on src/components/MTableEditCell/index.js:
```diff
@@ -72,61 +56,48 @@ function MTableEditCell(props) {
 
   const onApprove = () => {
-    setState({ ...state, isLoading: true });
+    setState({ ...state, isLoading: true }, () => {
```
setState from the useState does not have the callback which is in the class setState, so this does not work. Thats why it wa smoved into the useEffect which you removed.

Domino987 commented on src/components/MTableEditCell/index.js:
```diff
@@ -72,61 +56,48 @@ function MTableEditCell(props) {
             columnDef={props.columnDef}
             value={state.value}
-            onChange={(prevState, value) => setState({ ...prevState, value })}
+            onChange={(value) => setState({ value })}
```
This spreading is also needed since you would override the loading state, This works for class but not for function components.

Domino987 commented on src/components/MTableEditCell/index.js:
```diff
@@ -72,61 +56,48 @@ function MTableEditCell(props) {
           />
         </div>
-        {renderActions()}
+        {state.isLoading && (
```
Nice

Domino987 reviewed (changes requested):
Hi thank you for the work. I finally got a bette look at it and I see a few problems from the refactor. Can you check them out?

Commit:
updated CHANGELOG.md

Commit:
add package lock

Commit:
updated CHANGELOG.md

Commit:
Release v2.3.38

dependabot[bot] committed:
Bump url-parse from 1.4.7 to 1.5.1

Bumps [url-parse](https://github.com/unshiftio/url-parse) from 1.4.7 to 1.5.1.
- [Release notes](https://github.com/unshiftio/url-parse/releases)
- [Commits](https://github.com/unshiftio/url-parse/compare/1.4.7...1.5.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump lodash from 4.17.20 to 4.17.21

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.20 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.20...4.17.21)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump hosted-git-info from 2.8.8 to 2.8.9

Bumps [hosted-git-info](https://github.com/npm/hosted-git-info) from 2.8.8 to 2.8.9.
- [Release notes](https://github.com/npm/hosted-git-info/releases)
- [Changelog](https://github.com/npm/hosted-git-info/blob/v2.8.9/CHANGELOG.md)
- [Commits](https://github.com/npm/hosted-git-info/compare/v2.8.8...v2.8.9)

Signed-off-by: dependabot[bot] <support@github.com>

matthewoestreich committed:
Update package.json

remove deps that were moved to a diff package for exporting

matthewoestreich committed:
Update package.json

matthewoestreich committed:
Merge pull request #195 from material-table-core/dependabot/npm_and_yarn/url-parse-1.5.1

Bump url-parse from 1.4.7 to 1.5.1

matthewoestreich committed:
Merge pull request #196 from material-table-core/dependabot/npm_and_yarn/lodash-4.17.21

Bump lodash from 4.17.20 to 4.17.21

matthewoestreich committed:
Merge pull request #197 from material-table-core/dependabot/npm_and_yarn/hosted-git-info-2.8.9

Bump hosted-git-info from 2.8.8 to 2.8.9

matthewoestreich committed:
Update index.d.ts

fix import typo

matthewoestreich committed:
updated CHANGELOG.md

matthewoestreich committed:
Release v2.3.39

matthewoestreich committed:
Update CHANGELOG.md

MrWittman612 committed:
rebase commit

MrWittman612 commented:
I think I did something wrong. I fetched upstream then rebased my refactor branch but I think I messed that up so.... I have not touched or look at this in so long that it might be just a good idea to scrap this, and if you want me to go through the files one by one and refactor them to functional components, do so on a different branch and merge request. You can reach out to me any time.

Domino987 commented:
@MrWittman612 Yes looks like some errors happened. How much work was it to generate these changes?
